iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergSplit.java

```diff
@@ -68,6 +73,23 @@ public Path getPath() {
     return new Path(tableLocation);
   }
 
+  @Override
+  public byte[] getBytesForHash() {
+    Collection<FileScanTask> fileScanTasks = innerSplit.task().files();
+
+    try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+      for (FileScanTask task : fileScanTasks) {
+        baos.write(task.file().path().toString().getBytes());
+        byte[] startBytes = new byte[Long.BYTES];
+        SerDeUtils.writeLong(startBytes, 0, task.start());
+        baos.write(startBytes);
+      }
+      return baos.toByteArray();
+    } catch (IOException ioe) {
+      throw new RuntimeException("Couldn't produce hash input bytes for HiveIcebergSplit: " + this, ioe);
```

Comment:
Would the entire query fail if the hash computation doesn't work here? If so, do we want to consider just ERROR logging it but proceeding with the execution even if it's slower due to suboptimal cache affinity? What do you think?

Reply:
Hmm I don't think we want to prepare for such cases. If the hash input calculation fails here then we probably have some corrupted split that shouldn't exist anyway.

Reply:
Okay, makes sense